src/views/RSVP.js

```diff
@@ -0,0 +1,220 @@
+    e.preventDefault();
+
+    this.setState(
+      {
+        formErrors: {
+          firstName: this.formRefs.firstName.value ? null : "Required",
+          lastName: this.formRefs.lastName.value ? null : "Required",
+          email: this.formRefs.email.value
+            ? !this.formRefs.email.value.match(
+                /^\w[.!#$%&'*+-/=?^_`{|}~\w]*@[-\w]+\.[\w]+$/gi
+              )
+              ? "Invalid E-mail"
+              : null
+            : "Required"
+        }
+      },
+      () => {
+        if (
+          Object.keys(this.state.formErrors).some(
+            input => this.state.formErrors[input]
+          )
+        ) {
+          console.log("Error(s) found in form");
+        } else {
+          console.log("Form submitted!");
+        }
+      }
+    );
+  };
+
+  render() {
+    return (
+      <Main>
+        <Article left view="0">
+          <img src="/img/photos/stock_blue_flowers.png" alt="Blue Flowers" />
+        </Article>
+        <Article right view="0" wide={true}>
+          <Heading
+            size="3.75rem"
+            top="2em"
+            left="0.8em"
+            width="calc(100% - 0.8em)"
+          >
+            LET'S CELEBRATE!
+          </Heading>
+          <StyledRSVPForm>
+            <FormArea
+              width="15.125em"
+              label="FIRST NAME"
+              error={this.state.formErrors.firstName}
+            >
+              <StyledFormInput
+                type="text"
+                ref={node => (this.formRefs.firstName = node)}
+                error={this.state.formErrors.firstName}
+              />
+            </FormArea>
+            <FormArea
+              width="15.125em"
+              float="right"
+              label="LAST NAME"
+              error={this.state.formErrors.lastName}
+            >
+              <StyledFormInput
+                type="text"
+                ref={node => (this.formRefs.lastName = node)}
+                error={this.state.formErrors.lastName}
+              />
+            </FormArea>
+            <FormArea
+              label="E-MAIL ADDRESS"
+              error={this.state.formErrors.email}
+            >
+              <StyledFormInput
+                type="email"
+                ref={node => (this.formRefs.email = node)}
+                error={this.state.formErrors.email}
+              />
+            </FormArea>
+            <FormArea label="ACCOMMODATIONS">
+              <StyledFormTextArea
+                type="text"
+                ref={node => (this.formRefs.accommodations = node)}
+              />
+            </FormArea>
+            <FormArea label="MESSAGE TO THE COUPLE">
+              <StyledFormTextArea
+                type="text"
+                ref={node => (this.formRefs.message = node)}
+              />
+            </FormArea>
+            <StyledFootnote>KINDLY RESPOND BY OCTOBER 5, 2019</StyledFootnote>
+            <StyledSubmitButton onClick={this.handleOnSubmit}>
```

Comment:
It's more preferred to have an `onSubmit` prop on the `<form />` element, and set it to `this.handleOnSubmit`.

```jsx
<form onSubmit={this.handleOnSubmit}>
  { /* ... */ }
</form>
```

Add a `type="submit"` to the `<StyledSubmitButton />` component.